modules/sfp_stackoverflow.py

```diff
@@ -0,0 +1,257 @@
+                self.error(f"Error querying StackExchange API: {e}")
+                self.errorState = True
+                return None
+
+        # Questions profile endpoint, used to return displayname
+        if qryType == "questions":
+            try:
+                res = self.sf.fetchUrl(
+                    f"https://api.stackexchange.com/2.3/questions/{qry}?order=desc&sort=activity&site=stackoverflow",
+                    timeout=self.opts['_fetchtimeout'],
+                    useragent="SpiderFoot"
+                )
+            except Exception as e:
+                self.error(f"Error querying StackExchange API: {e}")
+                self.errorState = True
+                return None
+
+        if res['content'] is None:
+            self.info(f"No Stackoverflow info found for {qry}")
+            return None
+
+        try:
+            return json.loads(res['content'])
+        except Exception as e:
+            self.error(f"Error processing JSON response from Stackoverflow: {e}")
+            self.errorState = True
+            return None
+
+    def extractUsername(self, questionId):
+        # Need to query the questions endpoint with the question_id to find the username
+        query_results = self.query(questionId, "questions")
+
+        items = query_results.get('items')
+
+        if items is None:
+            return None
+
+        for item in items:
+            owner = item['owner']
+            username = owner.get('display_name')
+
+        return str(username)
+
+    def extractIP4s(self, text):
+        ips = set()
+
+        matches = re.findall(r'^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$', text)
+
+        if matches:
+            for match in matches:
+                if self.sf.validIP(match) and not(self.sf.isValidLocalOrLoopbackIP(match)):
+                    ips.add(match)
+            return list(ips)
+        else:
+            return None
+
+    def extractIP6s(self, text):
+        ips = set()
+
+        matches = re.findall(r'(?:^|(?<=\s))(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))(?=\s|$)', text)
+
+        if matches:
+            for match in matches:
+                if self.sf.validIP6(match) and not(self.sf.isValidLocalOrLoopbackIP(match)):
+                    ips.add(match)
+            return list(ips)
+        else:
+            return None
+
+    def handleEvent(self, event):
```

Comment:
For all events reported, there should be some checking to avoid reporting the same event twice. For example, testing with `tesla.com` I'm seeing the same email address being reported multiple times.